File: javatest/src/main/java/com/example/proxy/java/TestProxy.java
package com.example.proxy.java;

/**
 *
 * Created by jaylim on 2017/7/24.
 */
public class TestProxy {
    public static void main(String args[]) {
        UserServiceImpl userServiceImpl = new UserServiceImpl();
        MyInvocationHandler handler = new MyInvocationHandler();
        Service service = (Service) handler.bind(userServiceImpl);
        service.add();
    }
}

File: javatest/src/main/java/com/example/proxy/TestProxy.kt
package com.example.proxy

/**
 *
 * Created by jaylim on 2017/7/24.
 */
object TestProxy {
    @JvmStatic fun main(args: Array<out String>) {
        val serviceImpl = UserServiceImpl()
        val invocationHandler = MyInvocationHandler()
        val bind = invocationHandler.bind(serviceImpl) as Service
        bind.add("jinglm")
    }
}